src/service/ampdoc-impl.js

```diff
@@ -422,21 +422,31 @@ export class AmpDoc {
   /**
    * Returns whether the specified extension has been declared on this ampdoc.
    * @param {string} extensionId
+   * @param {string=} opt_version
    * @return {boolean}
    */
-  declaresExtension(extensionId) {
-    return this.declaredExtensions_.indexOf(extensionId) != -1;
+  declaresExtension(extensionId, opt_version) {
+    const declared = this.declaredExtensions_[extensionId];
+    if (!declared) {
+      return false;
+    }
+    return !opt_version || declared === opt_version;
   }
 
   /**
    * Adds a declared extension to an ampdoc.
    * @param {string} extensionId
+   * @param {string} version
    * @restricted
    */
-  declareExtension(extensionId) {
-    if (!this.declaresExtension(extensionId)) {
-      this.declaredExtensions_.push(extensionId);
-    }
+  declareExtension(extensionId, version) {
+    devAssert(version); // QQQ: remove/debugging.
+    devAssert(
+      (this.declaredExtensions_[extensionId] || version) === version,
```

Comment:
Hm?

Reply:
This is just a simple dev check to ensure that two different versions are not registered anywhere by accident since we didn't enforce this limitation before. Helps with tests and future work. So I'd rather like to keep it.

Reply:
I think it's just the way that it's written. It allows redeclaration if the version matches, or there's declaration. Would it make more sense to write:

```suggestion
      !this.declaredExtensions_[extensionId] || this.declaredExtensions_[extensionId] === version,
```

Reply:
Oh. Sure. Done.